### ISSUE TITLE
`m'` doesn't save position to jump list

### ISSUE THREAD

**Issue by infokiller** [neovim/neovim#11054]
I tried both `m'` and `` m` ``. I tried older versions and the issue was introduced between `0.3.8` and `0.4.0` (didn't try in between).

- `nvim --version`:
```
NVIM v0.4.2
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -O2 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/build/neovim/src/build/config -I/build/neovim/src/neovim-0.4.2/src -I/usr/include -I/build/neovim/src/build/src/nvim/auto -I/build/neovim/src/build/include
Compiled by builduser

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```
- `vim -u DEFAULTS` (version: ) behaves differently?
Yes
- Operating system/version: Archlinux 5.2.14
- Terminal name/version: Termite in Tmux
- `$TERM`: `screen-256color`

### Steps to reproduce using `nvim -u NORC`

```bash
nvim -u NORC --cmd 'set shada=' -c 'call feedkeys("\<Down>m`:jumps\<CR>", "t")' <(echo '1\n2\n')
```

### Actual behaviour

Line 2 not in the jump list:

![image](https://user-images.githubusercontent.com/10516814/65166220-f84ff180-da48-11e9-98f2-800cff847f3e.png)

### Expected behaviour

Line 2 in the jump list (screenshot from `0.3.8`):

![image](https://user-images.githubusercontent.com/10516814/65166274-1289cf80-da49-11e9-8bd0-354b0d766963.png)

**Comment by blueyed** [comment]
Bisected to 35362495c965554f45634bcde3c4ce6d5eca52aa:

> jumplist: avoid extra tail entry #9805
> fixes #9775

/cc @abdelhakeem

**Comment by justinmk** [comment]
This is a duplicate and already discussed.
